miden-lib/asm/kernels/transaction/lib/tx.masm

```diff
@@ -178,6 +184,54 @@ proc.add_non_fungible_asset_to_note
     # => [note_ptr, note_idx]
 end
 
+#! Updates the transaction expiration block number.
+#!
+#! The input block_height_delta is added to the block reference number in order to output an upper
+#! limit at which the transaction will be considered valid (not expired).
+#! This value can be later decreased, but not increased. 
+#!
+#! Inputs: [block_height_delta, ...]
+#! Output: [...]
+#!
+#! Where:
+#! - block_height_delta is the desired expiration time delta (1 to 0xFFFF).
+#! - tx_block_height_delta is the updated (or unchanged) transaction expiration time delta.
+export.update_expiration_block_num
+    # Ensure block_height_delta is between 1 and 0xFFFF (inclusive)
+    dup neq.0 assert.err=ERR_INVALID_TX_EXPIRATION_DELTA
+    dup push.EXPIRY_UPPER_LIMIT lt assert.err=ERR_INVALID_TX_EXPIRATION_DELTA
+    # => [block_height_delta]
+
+    exec.get_block_number add
+    # => [absolute_expiration_num]
+
+    # Load the current stored delta from memory
+    dup exec.memory::get_expiration_block_num  
+    # => [block_height_delta, absolute_expiration_num, absolute_expiration_num]
+
+    # Check if block_height_delta is greater
+    lt
```

Comment:
Since we know that both `stored_expiration_block_num ` and `absolute_expiration_num` are `u32` values, we can use `u32lt` instruction here which is more efficient (3 cycles vs 17).

Reply:
I think `absolute_expiration_num` can be above `u32::MAX` by at most `0xFFFF` once the block ref gets close enough. This would likely not happen for a good number of years, but I changed the `add` to an `u32overflowing_add`. If it overflows, no need to continue. Though there might be better ways of doing this.

Reply:
Not sure if we need to handle this edge case as by the time we get to this situation we are basically just a few days away from the "end of time" (e.g., before we get to max possible block number) - and this is probably more than 100 years out. Hopefully, by that time, this will be solved in some other ways.